artifactory/commands/python/dependencies/dependencies.go

```diff
@@ -60,58 +59,46 @@ func UpdateDepsChecksumInfo(dependenciesMap map[string]*buildinfo.Dependency, ca
 
 // Before running this function, dependency IDs may be the file names of the resolved python packages.
 // Update build info dependency IDs and the requestedBy field.
-// dependenciesMap   - Dependency name to Dependency map
-// dependenciesGraph - Dependency graph as built by 'pipdeptree' or 'pipenv graph'
-// packageName       - The resolved package name of the Python project, may be empty if we couldn't resolve it
-// moduleName        - The input module name from the user, or the packageName
-func UpdateDepsIdsAndRequestedBy(dependenciesMap map[string]*buildinfo.Dependency, dependenciesGraph map[string][]string, packageName, moduleName string) {
-	// Create a map from dependency to parents
-	requestedByMap := make(map[string][]string)
-	for parent, children := range dependenciesGraph {
-		for _, child := range children {
-			if packageName == "" || !strings.HasPrefix(parent, packageName+":") {
-				requestedByMap[child] = append(requestedByMap[child], parent)
+// allDependencies      - Dependency name to Dependency map
+// dependenciesGraph    - Dependency graph as built by 'pipdeptree' or 'pipenv graph'
+// topLevelPackagesList - The direct dependencies
+// packageName          - The resolved package name of the Python project, may be empty if we couldn't resolve it
+// moduleName           - The input module name from the user, or the packageName
+func UpdateDepsIdsAndRequestedBy(allDependencies map[string]*buildinfo.Dependency, dependenciesGraph map[string][]string,
+	topLevelPackagesList []string, packageName, moduleName string) {
+	if packageName == "" {
```

Comment:
I think that when packageName == "" you can add an entry to the dependenciesGraph like this:
dependenciesGraph[moduleName] = []string{...topLevelPackagesList}
and then run the same flaw as for packageName != ""